src/app/modules/verification/verification.controller.ts

```diff
@@ -168,6 +175,81 @@ export const handleGetOtp: RequestHandler<
     })
 }
 
+/**
+ * NOTE: This is exported solely for testing
+ * Generates an otp when a user requests to verify a field.
+ * The current answer is signed, and the signature is also saved in the transaction, with the field id as the key.
+ * @param answer The mobile or email number of the user
+ * @param transactionId The id of the transaction to verify
+ * @param formId The id of the form to verify
+ * @param fieldId The id of the field to verify
+ * @returns 201 when otp generated successfully
+ * @returns 400 when the parameters could not be parsed
+ * @returns 400 when the transaction has expired
+ * @returns 400 when the otp could not be sent via sms
+ * @returns 400 when the otp could not be sent via email
+ * @returns 400 when the provided phone number is not valid
+ * @returns 400 when the field type is not supported for validation
+ * @returns 404 when the requested form was not found
+ * @returns 404 when the transaction was not found
+ * @returns 404 when the field was not found
+ * @returns 422 when the user requested for a new otp without waiting
+ * @returns 500 when the otp could not be hashed
+ * @returns 500 when there is a database error
+ */
+export const _handleGenerateOtp: RequestHandler<
+  { transactionId: string; formId: string; fieldId: string },
+  ErrorDto,
+  { answer: string }
+> = async (req, res) => {
+  const { transactionId, formId, fieldId } = req.params
+  const { answer } = req.body
+  const logMeta = {
+    action: 'handleGenerateOtp',
+    transactionId,
+    fieldId,
+    ...createReqMeta(req),
+  }
+  return FormService.retrieveFormById(formId)
+    .andThen(() => generateOtpWithHash(logMeta, SALT_ROUNDS))
+    .andThen(({ otp, hashedOtp }) =>
+      VerificationFactory.sendNewOtp({
+        fieldId,
+        hashedOtp,
+        otp,
+        recipient: answer,
+        transactionId,
+      }),
+    )
+    .map(() => res.sendStatus(StatusCodes.CREATED))
+    .mapErr((error) => {
+      logger.error({
+        message: 'Error creating new OTP',
+        meta: logMeta,
+        error,
+      })
+      const { errorMessage, statusCode } = mapRouteError(error)
+      return res.status(statusCode).json({ message: errorMessage })
+    })
+}
+
+/**
+ * Handler for the POST /forms/:formId/fieldverifications/:transactionId/fields/:fieldId/otp/generate endpoint
+ */
+export const handleGenerateOtp = [
+  celebrate({
+    [Segments.BODY]: Joi.object({
+      fieldType: Joi.string().valid(...VERIFIED_FIELDTYPES),
```

Comment:
Not a fan of having extra keys that the controller does not actually need, just let the controller handle malformed data in this case, so we remove all the complexity of the conditional. We can still only allow the answer to be either an email or phone number string

Reply:
can i clarify on this point? my understanding of using `joi` is to ensure that the shape of the request adheres to what we expect. 

wouldn't this be better on the validator itself so that the controller can be ignorant of this (i.e., that the respective field type is valid) and avoid bringing in validation of the _request_ into the controller?

Reply:
fair enough, but i'd argue that this is just a guard right, since for example

lets say `fieldIdA` is a `BasicField.Mobile`

but you pass in `fieldType: BasicField.Email` and a valid email string into the POST body.

the controller will still need to validate that the fieldId is indeed a mobile field and thus the arguments are invalid. Joi validation does nothing for this case, and thus what is the point?

Reply:
@karrui @mantariksh agreed, i think i'll change it so that the validation takes place on the controller rather than within the service itself.

this allows us to introduce [nominal typing](answer) in our codebase with the help of [ts-brand](answer), as valid contact numbers/emails aren't just any string but rather, special instances of strings that have passed the validation upfront. this also allows us to parse rather than validate in that we are preserving information in the type system and have it be correct _by definition_ rather than validate it at the call site

Reply:
is that necessary for this PR? might it be better to preserve existing functionality in this PR and introduce your suggested changes separately, so we can separately evaluate the impact of the new validation without coupling it to this PR (which is meant to be a refactor)?

Reply:
@mantariksh good point, that slipped my mind - will make the appropriate changes here and introduce suggestions in another PR!